File: 2022-06-craft-conf-languages/examples/download.go
package main

func Download(url string) error { return nil }

func DownloadFiles() error {
	err := Download("https://a.example.com/a.csv")
	if err != nil {
		return err
	}

	err = Download("https://b.example.com/b.csv")
	if err != nil {
		return err
	}

	err = Download("https://c.example.com/c.csv")
	if err != nil {
		return err
	}

	return nil
}

func main() {
	DownloadFiles()
}

File: 2022-06-craft-conf-languages/examples/download.rs
#[derive(Debug)]
struct Error;

async fn download(url: &str) -> Result<(), Error> {
    Ok(())
}

async fn download_files() -> Result<(), Error> {
    futures::try_join!(
        download("https://a.example.com/a.csv"),
        download("https://b.example.com/b.csv"),
        download("https://c.example.com/c.csv"),
    )?;
    Ok(())
}

async fn main() {
    download_files().await.unwrap();
}
